### PR TITLE
Refactoring for compactness, and taking advantage of chaining with Re…

### PR DESCRIPTION
**Pull request by mathomas**
…sultActions.

I like to refactor code I'm trying to learn.  As such, this doesn't have to be merged, but I'd like some feedback on simplification/terseness.

- Variables holding Mocks don't have to be called "mock".  It's repetitive, and they damn well better be, in a test.
- Static imports rule.
- ResultActions supports chaining, and thus nice tight expressions of expectations.
- Noticing Kotlin doesn't have something like `private static`, but `private val` works OK.  Naming?  Kotlin also doesn't seem to like all-caps.  Boo.

### CODE REVIEW

File: src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt [hunk 1]
```diff
@@ -19,159 +19,131 @@ import org.junit.jupiter.api.extension.ExtendWith
...
         @Test
-        fun getExistingBookingTest() {
-            // arrange
-            // act
-            val result = mockMvc.perform(MockMvcRequestBuilders.get("/v1/booking/1"))
-
-            // assert
-            result.andExpect(MockMvcResultMatchers.status().isOk)
-            result.andExpect(MockMvcResultMatchers.jsonPath<String>("$.subject", Matchers.equalToIgnoringCase("The Booking")))
+        fun existingBooking() {
```
Comment (aochsner) [inline src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt]: eventually i plan to even do:
```
fun `get existing booking`() {
}
```

File: src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt [hunk 2]
```diff
@@ -19,159 +19,131 @@ import org.junit.jupiter.api.extension.ExtendWith
...
 import org.springframework.test.web.servlet.MockMvc
-import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
-import org.springframework.test.web.servlet.result.MockMvcResultMatchers
-import java.time.LocalDateTime
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import java.time.LocalDateTime.now
```
Comment (aochsner) [inline src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt]: slightly hesitant about this one only cause we have to be very persnickety that it's LocalDateTime...for now its fine cause I'd probably rather be consistent

File: src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt [hunk 3]
```diff
@@ -19,159 +19,131 @@ import org.junit.jupiter.api.extension.ExtendWith
...
  */
 @ExtendWith(SpringExtension::class)
 @WebMvcTest(BookingController::class)
 class BookingControllerMockMvcTests @Autowired constructor(
-    private val mockMvc: MockMvc,
+    private val mvc: MockMvc,
     private val mapper: ObjectMapper
 ) {
+    private val NYC = "America/New_York"
+    private val NYC_TZ = ZoneId.of(NYC)
```
Comment (aochsner) [inline src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt]: if you really wanted to learn re: static fields https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#static-fields

File: src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt (hunk 2 above)
Comment (mathomas) [reply src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt]: Good point.  I think we can save fully-qualified imports for the oddball cases, unless it becomes confusing.

File: src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt (hunk 3 above)
Comment (mathomas) [reply src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt]: Heh.  That site does not even begin to layout in Safari.  I did try, but not so hard as to open in a different browser :-D.  Looks like I'll be applying `const1` in the next round.

File: src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt (hunk 1 above)
Comment (mathomas) [reply src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt]: Totally!  I'll change in next round.

**Pull request merged**